components/category/Category.js

```diff
@@ -20,17 +41,31 @@ export default function Category({ category }) {
         {category.name}
       </CategoryHeadline>
       {isExtended && (
-        <CategoryList>
-          {listItems.map((listItem) =>
-            listItem.categoryId === category.id ? (
-              <CategoryListItem key={listItem.id} id={listItem.id}>
-                {listItem.name}
-              </CategoryListItem>
-            ) : (
-              ""
-            )
-          )}
-        </CategoryList>
+        <DragDropContext onDragEnd={handleOnDragEnd}>
+          <Droppable droppableId={category.id}>
+            {(provided) => (
+              <CategoryList
+                ref={provided.innerRef}
+                {...provided.droppableProps}
+              >
+                {listItems.map((listItem, index) =>
+                  listItem.categoryId === category.id ? (
+                    <CategoryListItem
+                      key={listItem.id}
+                      id={listItem.id}
+                      index={index}
```

Comment:
You are basically using a filtered list here, so you are jumping over some indizes here. Does the dnd library still behave well?

Reply:
Also this is the place where you are using "Strategy no. 1": Using the actual index of an item in the list.